flang/lib/Lower/CharacterExpr.cpp

```diff
@@ -21,8 +21,10 @@ static fir::CharacterType getCharacterType(mlir::Type type) {
     return boxType.getEleTy();
   if (auto refType = type.dyn_cast<fir::ReferenceType>())
     type = refType.getEleTy();
-  if (auto seqType = type.dyn_cast<fir::SequenceType>())
+  if (auto seqType = type.dyn_cast<fir::SequenceType>()) {
+    assert(seqType.getShape().size() == 1 && "rank must be 1");
```

Comment:
Why can't this function be used on character arrays ?

Reply:
It appears that it is not tested anywhere.

There was no checks for types that are arrays of CHARACTERs anywhere, which becomes a problem in areas like I/O, where the handling of a scalar CHARACTER is distinct from the handling of an array of CHARACTER even though each looks similar in FIR.